### PR TITLE
Remove redundant interfaces

### LINKED ISSUE #869: Remove redundant interfaces
Currently, each service has its interface, i.e. GameService is the interface and GameServiceImpl is its implementation. I never felt like this was a smart thing to do but it was and still is [recommended by the Spring Framework](https://docs.spring.io/spring/docs/current/spring-framework-reference/core.html#aop-proxying) ("JDK dynamic proxies are preferred whenever you have a choice").

Still not seeing any benefit after years, but seeing the unnecessary noise it causes, I want to get rid of all service interfaces that only have one "Impl"-implementation.

There are some classes who don't just have the "Impl" implementation, but also a Mock implementation. These can be omitted for now.

# Caveat

I'm not sure how Git will see the change if we remove the interfaces and rename the classes to what used to be the interface's name. It will probably not see rename operations like `GameServiceImpl -> GameService` but the deletion of `GameServiceImpl` and a change of all lines of `GameService`. I don't want the implementor of this issue to become the author of almost every line of every service ;)

Therefore, I recommend doing this in two steps:

1. Delete all service interfaces that only have one implementation, commit
1. Remove the "Impl" suffix of all affected classes, commit

### PR DESCRIPTION
**Pull request by micheljung**
Fixes #869

### CODE REVIEW

File: .idea/modules.xml [hunk 1]
```diff
@@ -2,9 +2,7 @@
 <project version="4">
   <component name="ProjectModuleManager">
     <modules>
-      <module fileurl="file://$PROJECT_DIR$/.idea/modules/downlords-faf-client.iml" filepath="$PROJECT_DIR$/.idea/modules/downlords-faf-client.iml" />
-      <module fileurl="file://$PROJECT_DIR$/.idea/modules/downlords-faf-client_main.iml" filepath="$PROJECT_DIR$/.idea/modules/downlords-faf-client_main.iml" group="downlords-faf-client" />
-      <module fileurl="file://$PROJECT_DIR$/.idea/modules/downlords-faf-client_test.iml" filepath="$PROJECT_DIR$/.idea/modules/downlords-faf-client_test.iml" group="downlords-faf-client" />
+      <module fileurl="file://$PROJECT_DIR$/downlords-faf-client.iml" filepath="$PROJECT_DIR$/downlords-faf-client.iml" />
```
Comment (Geosearchef) [inline .idea/modules.xml:5]: why do you change this (was probably your IDE, not you)

File: .idea/runConfigurations/FAF_Client_Local.xml [hunk 2]
```diff
@@ -1,22 +1,15 @@
 <component name="ProjectRunConfigurationManager">
   <configuration default="false" name="FAF Client Local" type="Application" factoryName="Application" singleton="true">
-    <extension name="coverage" enabled="false" merge="false" sample_coverage="true" runner="idea">
+    <option name="MAIN_CLASS_NAME" value="com.faforever.client.FafClientApplication" />
+    <module name="downlords-faf-client" />
+    <option name="VM_PARAMETERS" value="-XX:+UseConcMarkSweepGC -DnativeDir=build/resources/native -Dspring.profiles.active=local -Dprism.dirtyopts=false -Xmx512m -XX:+HeapDumpOnOutOfMemoryError" />
+    <option name="WORKING_DIRECTORY" value="file://$PROJECT_DIR$" />
+    <extension name="coverage">
```
Comment (Geosearchef) [inline .idea/runConfigurations/FAF_Client_Local.xml:7]: why move?

File: .idea/runConfigurations/FAF_Client_Offline.xml [hunk 3]
```diff
@@ -1,22 +1,15 @@
 <component name="ProjectRunConfigurationManager">
   <configuration default="false" name="FAF Client Offline" type="Application" factoryName="Application" singleton="true">
-    <extension name="coverage" enabled="false" merge="false" sample_coverage="true" runner="idea">
+    <option name="MAIN_CLASS_NAME" value="com.faforever.client.FafClientApplication" />
+    <module name="downlords-faf-client" />
+    <option name="VM_PARAMETERS" value="-XX:+UseConcMarkSweepGC -DnativeDir=build/resources/native -Dspring.profiles.active=offline -Dprism.dirtyopts=false -Xmx512m -XX:+HeapDumpOnOutOfMemoryError" />
+    <option name="WORKING_DIRECTORY" value="file://$PROJECT_DIR$" />
+    <extension name="coverage">
```
Comment (Geosearchef) [inline .idea/runConfigurations/FAF_Client_Offline.xml:7]: why move?

File: .idea/runConfigurations/FAF_Client_Prod.xml [hunk 4]
```diff
@@ -1,22 +1,15 @@
 <component name="ProjectRunConfigurationManager">
   <configuration default="false" name="FAF Client Prod" type="Application" factoryName="Application" singleton="true">
-    <extension name="coverage" enabled="false" merge="false" sample_coverage="true" runner="idea">
+    <option name="MAIN_CLASS_NAME" value="com.faforever.client.FafClientApplication" />
+    <module name="downlords-faf-client" />
+    <option name="VM_PARAMETERS" value="-XX:+UseConcMarkSweepGC -DnativeDir=build/resources/native -Dprism.dirtyopts=false -Xmx512m -XX:+HeapDumpOnOutOfMemoryError" />
+    <option name="WORKING_DIRECTORY" value="file://$PROJECT_DIR$" />
+    <extension name="coverage">
```
Comment (Geosearchef) [inline .idea/runConfigurations/FAF_Client_Prod.xml:7]: why move? (and change)

File: .idea/runConfigurations/FAF_Client_Test.xml [hunk 5]
```diff
@@ -1,22 +1,15 @@
 <component name="ProjectRunConfigurationManager">
   <configuration default="false" name="FAF Client Test" type="Application" factoryName="Application" singleton="true">
-    <extension name="coverage" enabled="false" merge="false" sample_coverage="true" runner="idea">
+    <option name="MAIN_CLASS_NAME" value="com.faforever.client.FafClientApplication" />
+    <module name="downlords-faf-client" />
+    <option name="VM_PARAMETERS" value="-XX:+UseConcMarkSweepGC -DnativeDir=build/resources/native -Dspring.profiles.active=test -Dprism.dirtyopts=false -Xmx512m -XX:+HeapDumpOnOutOfMemoryError" />
+    <option name="WORKING_DIRECTORY" value="file://$PROJECT_DIR$" />
+    <extension name="coverage">
```
Comment (Geosearchef) [inline .idea/runConfigurations/FAF_Client_Test.xml:7]: why move/change?

File: src/main/java/com/faforever/client/achievements/AchievementService.java [hunk 6]
```diff
@@ -2,22 +2,113 @@
...
 import com.faforever.client.api.dto.AchievementDefinition;
 import com.faforever.client.api.dto.PlayerAchievement;
+import com.faforever.client.config.CacheNames;
+import com.faforever.client.player.PlayerService;
+import com.faforever.client.remote.AssetService;
+import com.faforever.client.remote.FafService;
+import com.faforever.client.remote.UpdatedAchievementsMessage;
+import com.google.common.annotations.VisibleForTesting;
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
```
Comment (Geosearchef) [inline src/main/java/com/faforever/client/achievements/AchievementService.java:12]: did you click optimize imports before committing?

**Review by Geosearchef — CHANGES_REQUESTED** [review]
pls also reduce the number of commits to 1

**Comment by Geosearchef** [comment]
Also, pls read #869 -> Caveat

```
I'm not sure how Git will see the change if we remove the interfaces and rename the classes to what used to be the interface's name. It will probably not see rename operations like GameServiceImpl -> GameService but the deletion of GameServiceImpl and a change of all lines of GameService. I don't want the implementor of this issue to become the author of almost every line of every service ;)

Therefore, I recommend doing this in two steps:

    Delete all service interfaces that only have one implementation, commit
    Remove the "Impl" suffix of all affected classes, commit
```

File: .idea/modules.xml (hunk 1 above)
Comment (dknz2008) [reply .idea/modules.xml:5]: Didn't touch it, must have been IDE.

File: .idea/runConfigurations/FAF_Client_Offline.xml (hunk 3 above)
Comment (dknz2008) [reply .idea/runConfigurations/FAF_Client_Offline.xml:7]: Must have been IDE, no idea

File: src/main/java/com/faforever/client/achievements/AchievementService.java (hunk 6 above)
Comment (dknz2008) [reply src/main/java/com/faforever/client/achievements/AchievementService.java:12]: Just did that, and squashed the commits into one.  Problem is all the commits are still listed above (not sure how to remove them).

**Comment by micheljung** [comment]
Thanks! I rebased, squashed and merged it in 19c19ad and f5512ac

**Pull request closed without merging**